### ISSUE TITLE
Chinese (Pinyin): Properly handle partial suggestion words

### ISSUE THREAD

**Issue by tusooa** [maliit/keyboard#39]
On master

1. Switch language to zh-hans
2. Type `miaowu`
3. Press on any single character
4. Press Backspace

Observed:
in 3: the single character is inputted.
in 4: the character is removed.

Expected (ideal):
in 3: selected character kept in a buffer, displayed on the left. Suggestion list should be for the remaining input. For the example video, the buffer should be `喵wu`, and suggestions be those for `wu`.
in 4: return to the state just after 2. i.e. the Backspace serves roughly the same as an undo.

Video:
https://tusooa.xyz/_matrix/media/r0/download/tusooa.xyz/wPSiRWcURuzdzGkqFFLTrtHb

**Comment by jpetersen** [comment]
Thanks a lot for the reports. I am happy to get more improvement ideas.

**Comment by tusooa** [comment]
The problem is here:
```
src/plugin/inputmethod_p.h-
src/plugin/inputmethod_p.h:        QObject::connect(wordRibbon, &MaliitKeyboard::WordRibbon::wordCandidateSelected,
src/plugin/inputmethod_p.h-                         &editor,  &MaliitKeyboard::AbstractTextEditor::replaceAndCommitPreedit);
```

Whenever the user selects a candidate word, the whole preedit text is replaced with *that* word, and it is then unconditionally committed. This is problematic, as the normal Pinyin input process does not commit directly *until the whole sequence is turned into Chinese characters*.

**Comment by tusooa** [comment]
> Thanks a lot for the reports. I am happy to get more improvement ideas.

I got some work done in https://github.com/tusooa/keyboard/tree/tusooa/39-partial-suggestions . But I am not sure how to proceed.

Basically I want to do two things:

0. After a user selects a candidate word, refrain from committing, and only commit when the whole sequence is processed.
1. After a user selects a candidate word, if the sequence processing is not finished, *refresh* the candidate word list from the language plugin.

(0) is done. For (1), I tried to use AbstractLanguagePlugin::newPredictionSuggestions(), but that does not do the job as it does not clear previous candidates, and the candidate list is messed up with outdated candidates, which is not good.
